src/PKSim.Core/Snapshots/Mappers/ClassificationMapper.cs

```diff
@@ -8,56 +7,53 @@
 
 namespace PKSim.Core.Snapshots.Mappers
 {
-   public class ClassificationMapper : SnapshotMapperBase<ModelClassification, SnapshotClassification, object, ClassificationContext> 
+   public class ClassificationMapper : SnapshotMapperBase<ModelClassification, SnapshotClassification, ClassificationType, ClassificationContext>
    {
-      public override Task<ModelClassification> MapToModel(SnapshotClassification snapshot, object context)
+      public override Task<ModelClassification> MapToModel(SnapshotClassification snapshot, ClassificationType classificationType)
       {
-         throw new NotImplementedException();
+         var classification = new ModelClassification
+         {
+            ClassificationType = classificationType,
+            Name = snapshot.Name
+         };
+
+         return Task.FromResult(classification);
       }
 
       public override Task<SnapshotClassification> MapToSnapshot(ModelClassification model, ClassificationContext context)
       {
-         return mapTreeFrom(model, context.Classifications, context.Classifiables);
+         return mapTreeFrom(model, context);
       }
 
-      private async Task<SnapshotClassification> mapClassificationToSnapshot(ModelClassification model)
+      private async Task<SnapshotClassification> mapTreeFrom(ModelClassification classification, ClassificationContext context)
       {
-         return await SnapshotFrom(model, x =>
+         var root = await SnapshotFrom(classification, x =>
          {
-            x.Name = model.Name;
+            x.Name = classification.Name;
          });
-      }
-
-      private async Task<SnapshotClassification> mapTreeFrom(ModelClassification classification, IReadOnlyList<ModelClassification> classifications, IReadOnlyList<IClassifiableWrapper> classifiables)
-      {
-         var root = await mapClassificationToSnapshot(classification);
 
-         var childClassifications = childClassificationsFrom(classification, classifications);
-         root.Classifications = await mapChildClassifications(classifications, classifiables, childClassifications);
+         var childClassifications = childClassificationsFrom(classification, context.Classifications);
+         var childClassifiables = childClassifiablesFrom(classification, context.Classifiables);
 
-         root.Classifiables = childClassifiablesFrom(classification, classifiables).ToArray();
+         root.Classifications = childClassifications.Any() ? await Task.WhenAll(childClassifications.Select(x => mapTreeFrom(x, context))) : null;
+         root.Classifiables = childClassifiables.Any() ? childClassifiables.ToArray() : null;
```

Comment:
to array not required if the method returns an array (right now it returns  a list)